andela/travel_tool_back
# #162045684 Enable travel team members verify requests

#### What does this PR do?
Provides endpoints for travel team members to view approved requests and verify them

#### Description of Task to be completed?
- travel team members can retrieve list of approved requests for their centers
- travel team members can verify an approved request as travel ready
- update requests and approvals status enum column

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Enter into the project directory - `cd travel_tool_back`
- checkout the branch `ft-verify-request-162045684`
- Install dependencies - `yarn install`
- Rollback and run database migrations - `yarn db:rollmigrate`
- Start the server - `make start`
- Navigate to [travela](http://travela-local.andela.com:3000/#) in your browser and login
- Login on your browser and grab your token from the application tab in Developers tab
- Open Postman
- Add your token in headers
- Create a request using the following [guide](https://github.com/andela/travel_tool_back/pull/44) and set yourself as the approver for the request. (Ensure the origin of the trip is your current location)
- Approve the request as a manager using the following [guide](https://github.com/andela/travel_tool_back/pull/30)
- Nagivate to your database GUI (pgAdmin or postico)
- Update your user role to travel team member
- Make a get request to `localhost:5000/api/v1/approvals?verified=true` to retrieve requests pending verification
- Make a put request to `localhost:5000/api/v1/requests/<requestId>/verify` to verify a request
- An updated record will be returned with the status of "Verified"

#### Any background context you want to provide?
Ensure you have the privilege of a travel team member

#### What are the relevant pivotal tracker stories?
[#162045684](https://www.pivotaltracker.com/story/show/162045684)

#### Screenshots (if appropriate)
N/A

#### Questions:
